### ISSUE TITLE
ScreenQuad was used to render some images and crash occurred

### ISSUE THREAD

**Issue by ZhangGuangXin** [Rajawali/Rajawali#2077]
I used ScreenQuad, set the Material for it, and then set ATexture for the Material. On most mobile phones, there is no problem, but on OPPO R7, crash will appear. My code is as follows:：
    @Override
    protected void onRender(long ellapsedRealtime, double deltaTime) {
        try{
            super.onRender(ellapsedRealtime, deltaTime);   //The problem is this line of code

            if(isLoadOver){
                if(material != null && mFrameCount < maxFrameCount-1) {
                    material.getTextureList().remove(mTextures[mFrameCount]);
                    mFrameCount++;
                    material.getTextureList().add(mTextures[mFrameCount]);

                    if(mFrameCount == (maxFrameCount-1)){
                        openActivityPage();
                    }
                }
            }

            TimeUnit.MILLISECONDS.sleep(40);
        }catch (Exception e){
            e.printStackTrace();
        }
    }

The error message is as follows:
01-03 22:14:46.478 32567-1125/com.taobao.shoppingstreets E/AndroidRuntime: FATAL EXCEPTION: GLThread 1756
    Process: com.taobao.shoppingstreets, PID: 32567
    java.lang.IllegalAccessError: tried to access class org.rajawali3d.materials.shaders.AShaderBase$RVec3[] from class org.rajawali3d.materials.shaders.fragments.LightsVertexShaderFragment
        at org.rajawali3d.materials.shaders.fragments.LightsVertexShaderFragment.initialize(LightsVertexShaderFragment.java:114)
        at org.rajawali3d.materials.shaders.fragments.LightsVertexShaderFragment.<init>(LightsVertexShaderFragment.java:94)
        at org.rajawali3d.materials.Material.createShaders(Material.java:638)
        at org.rajawali3d.materials.Material.useProgram(Material.java:821)
        at org.rajawali3d.Object3D.render(Object3D.java:275)
        at org.rajawali3d.primitives.ScreenQuad.render(ScreenQuad.java:250)
        at org.rajawali3d.Object3D.render(Object3D.java:174)
        at org.rajawali3d.scene.Scene.render(Scene.java:1135)
        at org.rajawali3d.scene.Scene.render(Scene.java:1002)
        at org.rajawali3d.renderer.Renderer.render(Renderer.java:426)
        at org.rajawali3d.renderer.Renderer.onRender(Renderer.java:416)
        at com.taobao.shoppingstreets.ui.NormalArRender.onRender(NormalArRender.java:284)
        at org.rajawali3d.renderer.Renderer.onRenderFrame(Renderer.java:391)
        at org.rajawali3d.view.SurfaceView$RendererDelegate.onDrawFrame(SurfaceView.java:237)
        at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1579)
        at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1285)

Excuse me, what is the reason? What should I do?  I use the rajawali version 1.1.970

**Comment by contriteobserver** [comment]
You might want to consider `AnimatedGIFTexture`. Otherwise, this should get you there:
```
getTextureManager().replaceTexture(mTextures[mFrameCount]);
```

**Comment by ZhangGuangXin** [comment]
@contriteobserver Thank you for your reply. I solved the problem, but not in the way you said. The root cause of this problem is that I initialized ScreenQuad and gave it the Material, which is not in the opengl thread. Now I just put it in the onRender method, which solves the problem. But I was still very surprised, why in some phones, I don't do this, and there is no problem, but in some phones, there will be a problem.
